### PR TITLE
Product updates and CLI handling

### PR DESCRIPTION
**Pull request by maltesander**
## Description
This PR is part of a chain of PRs across all our projects to make them as consistent as possible in preparation for our first release.

Things it can contain (not all might be applicable to all repositories):

- Added CLI handling
- Cargo.toml
- Add repository key
- Set version properly
- Sort keys
- deny.toml consistency
- Pull request template
- Github CI consistency (e.g. rustdoc)
- cargo update
- Dependabot does not ignore stackable components anymore
- Adds changelog & readme
- Correct license

Fixes [61](https://github.com/stackabletech/nifi-operator/issues/61)

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: server/Cargo.toml [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
 edition = "2018"
 description = "Stackable Operator for Apache NiFi"
 license = "OSL-3.0"
+name = "stackable-nifi-operator-server"
+repository = "https://github.com/stackabletech/nifi-operator"
+version = "0.1.0-nightly"
 
 [dependencies]
-stackable_config = { git = "https://github.com/stackabletech/common.git", branch = "main" }
+stackable-config = { git = "https://github.com/stackabletech/common.git", branch = "main" }
```
Comment (lfrancke) [inline server/Cargo.toml:12]:
```suggestion
stackable-config = { git = "https://github.com/stackabletech/common.git", tag = "0.1.0" }
```

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: server/Cargo.toml (hunk 1 above)
Comment (maltesander) [reply server/Cargo.toml:12]: Its actually not even required. Ill remove it.

Comment (lfrancke) [reply server/Cargo.toml]: Even better

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: deny.toml [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
 
-# List of explictly allowed licenses
-# See https://spdx.org/licenses/ for list of possible licenses
-# [possible values: any SPDX 3.11 short identifier (+ optional exception)].
+[licenses]
+unlicensed = "deny"
+copyleft = "deny"
+allow-osi-fsf-free = "neither"
+default = "deny"
+confidence-threshold = 1.0
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply deny.toml:24]: (I can do so if you like, let me know)

**Review by lfrancke — APPROVED** [review]

**Pull request merged**